wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession+Clients.swift

```diff
@@ -66,24 +66,24 @@ extension ZMUserSession {
     /// - Returns: Token that needs to be stored as long the observer should be active.
 
     @objc(addClientUpdateObserver:)
-    public func addClientUpdateObserver(_ observer: ClientUpdateObserver) -> Any {
+    public func addClientUpdateObserver(_ observer: ClientUpdateObserver) -> NSObjectProtocol {
 
-        return ZMClientUpdateNotification.addObserver(context: managedObjectContext) { [weak self] (type, clientObjectIDs, error) in
+        return ZMClientUpdateNotification.addObserver(context: managedObjectContext) { [weak self, weak observer] type, clientObjectIDs, error in
```

Comment:
do we need to patch also all other observers ?

Reply:
I'm sure we have many more memory management issues, but in this case it led to wrong behavior and fixing it was the scope of this ticket and PR.